third_party/terraform/utils/node_config.go.erb

```diff
@@ -502,4 +509,58 @@ func flattenSandboxConfig(c *containerBeta.SandboxConfig) []map[string]interface
 	}
 	return result
 }
+
+func containerNodePoolLabelsSuppress(k, old, new string, d *schema.ResourceData) bool {
+	o, n := d.GetChange("node_config.0.sandbox_config.0.sandbox_type")
+	if o == nil || n == nil {
+		return false
+	}
+
+	if strings.HasPrefix(k, "node_config.0.labels.sandbox.gke.io/") {
+		return true
+	}
+
+	return false
+}
+
+func containerNodePoolTaintSuppress(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
I think we have a fairly similar function in resource_container_cluster: containerClusterAddedScopesSuppress. Can you look at that and see if using similar logic makes sense here?

Reply:
This logic makes _way_ more sense than the initial way I put this PR together. I have updated the code to use a similar strategy. Note that it does get called multiple times -- once for each "stringable" key/value Terraform provides -- but I think the performance hit is worth the clarity in this case. Now the suppression will be consistent across the entire field, rather than changing for some sub-keys (it is not clear at all what the upward propagation behavior is). Thank you for the pointer!